impeller/entity/geometry.cc

```diff
@@ -441,14 +521,58 @@ GeometryResult StrokePathGeometry::GetPositionBuffer(
   Scalar stroke_width = std::max(stroke_width_, min_size);
 
   auto& host_buffer = pass.GetTransientsBuffer();
-  auto vertex_buffer = CreateSolidStrokeVertices(
-      path_, host_buffer, stroke_width, miter_limit_ * stroke_width_ * 0.5,
-      stroke_cap_, GetJoinProc(stroke_join_), GetCapProc(stroke_cap_),
+  auto vertex_builder = CreateSolidStrokeVertices(
+      path_, stroke_width, miter_limit_ * stroke_width_ * 0.5, stroke_cap_,
+      GetJoinProc(stroke_join_), GetCapProc(stroke_cap_),
       entity.GetTransformation().GetMaxBasisLength());
 
   return GeometryResult{
       .type = PrimitiveType::kTriangleStrip,
-      .vertex_buffer = vertex_buffer,
+      .vertex_buffer = vertex_builder.CreateVertexBuffer(host_buffer),
+      .transform = Matrix::MakeOrthographic(pass.GetRenderTargetSize()) *
+                   entity.GetTransformation(),
+      .prevent_overdraw = true,
+  };
+}
+
+GeometryResult StrokePathGeometry::GetPositionUVBuffer(
+    Rect texture_coverage,
+    Matrix effect_transform,
+    const ContentContext& renderer,
+    const Entity& entity,
+    RenderPass& pass) {
+  if (stroke_width_ < 0.0) {
+    return {};
+  }
+  auto determinant = entity.GetTransformation().GetDeterminant();
+  if (determinant == 0) {
+    return {};
+  }
+
+  Scalar min_size = 1.0f / sqrt(std::abs(determinant));
+  Scalar stroke_width = std::max(stroke_width_, min_size);
+
+  auto& host_buffer = pass.GetTransientsBuffer();
+  auto stroke_builder = CreateSolidStrokeVertices(
+      path_, stroke_width, miter_limit_ * stroke_width_ * 0.5, stroke_cap_,
+      GetJoinProc(stroke_join_), GetCapProc(stroke_cap_),
+      entity.GetTransformation().GetMaxBasisLength());
+
+  VertexBufferBuilder<TextureFillVertexShader::PerVertexData> vertex_builder;
+  stroke_builder.IterateVertices(
```

Comment:
I took a bit of a shortcut here, because refactoring the stroke computation to handle either path felt a bit like a premature optimization